common/djangoapps/student/docs/decisions/0001-anonymous-user-id-generation.rst

```diff
@@ -0,0 +1,42 @@
+Anonymous User Id Generation
+--------------
+
+Status
+======
+
+Accepted
+
+Context
+=======
+
+The student app provides a mechanism to generate multiple anonymous ids for a
+student.  The anonymous ID can be independent of all courses or it can be
+course specific.  To generate the anonymous ID, we currently hash the user's
+``id`` with the Django ``SECRET_KEY`` and a course key if provided.  The
+mapping between the anonymous ID and user ``id`` are saved in the
+``AnonymousUserID`` table.
+
+As it stands, if the ``SECRET_KEY`` is rotated students would get new anonymous
+IDs starting immediately after rotation.  This can cause downstream issues
+where the IDs are output from the system.  For example, the IDs are in tracking
+data and could be used to track a user's activity through a course for research
+purposes.
+
+Decisions
+=========
+
+Once an anonymous ID is generated for a user in a particular LearningContext
+(either a course or some other unit of learning), it will remain that way even
+if the secret used to generate the ID changes.  For any context where an
+anonymous ID does not already exist, a new ID will be generated using the
+latest ``SECRET_KEY``.
```

Comment:
If we're adjusting the keys to be resilient in this manner, can we just go all the way to making these anonymous IDs randomly generated?

Reply:
@ormsbee that's an interesting point, the function that generates this key, has the option to not persist it.  In this case, it would give a new anonymous key, each time, instead of being consistent within a short enough time period.  I'm not sure what assumptions that would break and so would be reticent to make that change now.

Reply:
[puzzlement] I was under the impression that we are decoupling anonymous IDs from using `SECRET_KEY` altogether.  Only for backward compatibility reasons, we want to maintain old generated IDs (which were created with the `SECRET_KEY`). See the acceptance criteria in https://openedx.atlassian.net/browse/ARCHBOM-1645.

Or am I misinterpreting that this ADR is for a transitory phase, followed by another ADR/step that decouples the two?